### PR TITLE
Feature/custom margins

### PR DESCRIPTION
**Pull request by jmjordan**
## Issue Link :link:
 <ul>
   <li> Is this a bug fix or a feature? </li>
   <li> Does it break any existing functionality?</li>
</ul>

## Goals of this PR :tada:
 <ul>
   <li> Why is the change important? </li>
   <li> What does this fix? </li>
   <li> How far has it been tested? </li>
 </ul>
 
## How Has This Been Tested :mag:

Please let us know if you have tested your PR and if we need to reproduce the issues. Also, please let us know if we need any relevant information for running the tests.

<ul>
 <li> User Interface Testing </li>
 <li> Application Testing </li>
</ul>

## Test Configuration :space_invader:

<ul>
 <li> Xcode version: </li>
 <li> Device/Simulator </li>
 <li> iOS version || MacOSX version</li>
</ul>

## Things to check on :dart:

 - [ ] My Pull Request code follows the coding standards and styles of the project 
 - [ ] I have worked on unit tests and reviewed my code to the best of my ability 
 - [ ] I have used comments to make other coders understand my code better 
 - [ ] My changes are good to go without any warnings 
 - [ ] I have added unit tests both for the happy and sad path 
 - [ ] All of my unit tests pass successfully before pushing the PR 
 - [ ] I have made sure all dependent downstream changes impacted by my PR are working 

Manual testing was conducted on a few example cards. I created a new example card, but did not include it in my PR. I did not use UIEdgeInsets as the custom margin in my protocol since the layout simply centers the card within the collection and does not distinguish between left and right margins.

### CODE REVIEW

**Comment by badrinathvm** [comment]
https://github.com/jmjordan  Can you add an example in Example project to illustrate the usuage ?

**Comment by badrinathvm** [comment]
@jmjordan Can you add a example in Example project  to illustrate custom margins usage ?

File: Example/CardParts/CustomMarginsCardController.swift [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+//  CardParts_Example
+//
+//  Created by Jonathan Jordan on 12/3/19.
+//  Copyright © 2019 CocoaPods. All rights reserved.
+//
+
+import Foundation
+import CardParts
+
+class CustomMarginsCardController: CardPartsViewController, CustomMarginCardTrait {
```
Comment (croossin) [inline Example/CardParts/CustomMarginsCardController.swift]: Could you just move the `CustomMarginCardTrait` to an extension of `CustomMarginsCardController` like [this example](https://github.com/intuit/CardParts/blob/master/Example/CardParts/ThemedCardController.swift#L52).  Besides that the PR looks great - just tested it!  We will merge once this and @badrinathvm change is complete.

Comment (jmjordan) [reply Example/CardParts/CustomMarginsCardController.swift]: Thanks! The requested changes have been completed.

**Comment by croossin** [comment]
Thanks @jmjordan - I will merge and get a release out before the weekend! Thanks.

P.S.  Feel free to mention on any other features you would like to see from CardParts.

**Pull request merged**